core-java-modules/core-java-collections-list-4/src/test/java/com/baeldung/list/iteration/IterateListSimultaneousltUnitTest.java

```diff
@@ -0,0 +1,74 @@
+package com.baeldung.list.iteration;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.data.util.StreamUtils;
+
+public class IterateListSimultaneousltUnitTest {
+
+    List<String> names = List.of("John", "Nik", "Tom");
+    List<Integer> numbers = List.of(1, 2, 3);
+
+    @Test
+    public void givenTwoLists_whenProcessByZipping_thenProcessedSimultaneously() {
```

Comment:
_thenProcessedSimultaneously_ doesn't really express the outcome of the test - the outcome is that the data is joined from both collections.